devel/playdate-sdk/Portfile

```diff
@@ -0,0 +1,93 @@
+
+PortSystem          1.0
+
+name                playdate-sdk
+version             1.12.2
+revision            0
+categories          devel
+maintainers         @idleberg
+license             Restrictive
+
+description         Playdate SDK
+long_description    This package installs includes a rich set of APIs, a \
+                    compiler, and documentation for Playdate game development.
+
+homepage            https://play.date/dev/
+master_sites        https://download-keycdn.panic.com/playdate_sdk/
+distname            PlaydateSDK-${version}
+distfiles           ${distname}.zip
+
+checksums           rmd160  f0768a68c50b9d92029b562fff36e9235562de80 \
+                    sha256  42c793eb1e5d5251f3f1b8ac0cd1b1489c5f3e5a763b130c31205496d12389ce \
+                    size    30080744
+
+depends_extract     port:xar
+
+worksrcdir          ${name}-${version}
+
+extract {
+    system -W ${workpath} "mkdir -p ${worksrcpath}/pkg"
+    system -W ${workpath} "mkdir -p ${worksrcpath}/files"
+    system -W ${distpath} "unzip ${distpath}/${distfiles} -d ${worksrcpath}/pkg"
+    system -W ${worksrcpath}/pkg "${prefix}/bin/xar -xf ${worksrcpath}/pkg/PlaydateSDK.pkg -C ${worksrcpath}/files"
+    system -W ${worksrcpath}/files/PlaydateSDK.pkg "cat Payload | gunzip -dc | cpio -i"
+    system -W ${worksrcpath}/files/PlaydateSDK.pkg "cp -R PlaydateSDK ${worksrcpath}/${name}"
+}
+
+use_configure no
+
+build {}
+
+destroot {
+    # Binaries
+    copy -force ${worksrcpath}/${name}/bin/Playdate\ Simulator.app ${destroot}${applications_dir}
+    copy -force ${worksrcpath}/${name}/bin/pdc ${destroot}${prefix}/bin
+    copy -force ${worksrcpath}/${name}/bin/pdutil ${destroot}${prefix}/bin
+
+    # Libraries, API, and resources
+    foreach dir {"C_API" "CoreLibs" "Disk" "Resources"} {
+        xinstall -d "${destroot}${prefix}/share/${name}/${dir}"
+        system "cp -R \"${worksrcpath}/${name}/${dir}\" \"${destroot}${prefix}/share/${name}\""
+    }
+
+    # Documentation
+    foreach dir {"Designing for Playdate" "Inside Playdate"} {
+        xinstall -d "${destroot}${prefix}/share/doc/${name}/${dir}"
+        system "cp -R \"${worksrcpath}/${name}/${dir}\" ${destroot}${prefix}/share/doc/${name}"
+        copy -force "${worksrcpath}/${name}/${dir}.html" \
+            "${destroot}${prefix}/share/doc/${name}/${dir}.html"
+    }
+    copy -force "${worksrcpath}/${name}/PlaydateSDK.docset" \
+        "${destroot}${prefix}/share/doc/${name}"
+
+    # Examples
+    foreach dir [glob -type d -nocomplain -tails -directory "${worksrcpath}/${name}/Examples/" *] {
+        xinstall -d "${destroot}${prefix}/share/examples/${name}/${dir}"
+        system "cp -R \"${worksrcpath}/${name}/Examples/${dir}\" ${destroot}${prefix}/share/examples/${name}"
+    }
+
+    # Set permissions for the Playdate Simulator's data store
+    xinstall -d -m 777 "${destroot}${prefix}/share/Disk/Data"
+}
+
+post-activate {
+    system "export PLAYDATE_SDK_PATH=${prefix}/share/${name}"
+}
```

Comment:
this isn't going to "stick" once the user restarts a shell or something. So you likely should add `notes` and state there that this should be added to the users shell init scripts?

Reply:
> this isn't going to "stick" once the user restarts a shell or something. So you likely should add `notes` and state there that this should be added to the users shell init scripts?

Alternatively, the typical approach is to add symlinks to items in your new bin directory, ensuring the names are unique enough to not clash with other ports. So ideally they'd all have a unique prefix, like `playdate-`. (The target scripts/utils don't need to be renamed, but the symlinks in `${prefix}/bin` should definitely have the prefix.)

With that being said, if the targets can't/don't infer the location of `PLAYDATE_SDK_PATH`, it might be good to create wrapper scripts [in `${prefix}/bin`] instead, which set that environment variable first. It's a convention we use elsewhere as well, and it significantly improves the user experience.